zkvm/src/signer/prover.rs

```diff
@@ -0,0 +1,150 @@
+#![allow(non_snake_case)]
+
+use crate::signer::*;
+use crate::transcript::TranscriptProtocol;
+use curve25519_dalek::ristretto::RistrettoPoint;
+use curve25519_dalek::scalar::Scalar;
+use rand;
+
+#[derive(Copy, Clone)]
+pub struct Nonce(Scalar);
+pub struct NoncePrecommitment(Scalar);
```

Comment:
```suggestion
pub struct NoncePrecommitment([u8;32]);
```

we don't use it as a scalar, we only hash it to 32 pseudo-random bytes and compare them with what we get in the next step.

Reply:
I think it's good to change the hash output from being a scalar to a byte array, but the transcript `commit_bytes` function outputs `[u8;64]` so I will use that length of byte array instead.

Reply:
actually, implemented as `[u8;32]` after all.